src/main/java/org/apache/sysml/api/DMLScript.java

```diff
@@ -230,127 +439,81 @@ public static void main(String[] args)
-		ENABLE_DEBUG_MODE = false;
-		
-		boolean parsePyDML = false;
+		RUNTIME_PLATFORM oldrtplatform 	= rtplatform; //keep old rtplatform
+		ExplainType oldexplain 					= EXPLAIN; //keep old explain
+
+		Options options = createCLIOptions();
 		try
 		{
-			String fnameOptConfig = null; //optional config filename
-			String[] scriptArgs = null; //optional script arguments
-			boolean namedScriptArgs = false;
-			
-			for( int i=2; i<args.length; i++ )
-			{
-				if( args[i].equalsIgnoreCase("-explain") ) { 
-					EXPLAIN = ExplainType.RUNTIME;
-					if( args.length > (i+1) && !args[i+1].startsWith("-") )
-						EXPLAIN = Explain.parseExplainType(args[++i]);
-				}
-				else if( args[i].equalsIgnoreCase("-stats") ) {
-					STATISTICS = true;
-					if (args.length > (i + 1) && !args[i + 1].startsWith("-"))
-						STATISTICS_COUNT = Integer.parseInt(args[++i]);
-				}
-				else if ( args[i].equalsIgnoreCase("-exec")) {
-					rtplatform = parseRuntimePlatform(args[++i]);
-					if( rtplatform==null ) 
-						return false;
-				}
-				else if (args[i].startsWith("-config=")) // legacy
-					fnameOptConfig = args[i].substring(8).replaceAll("\"", "");
-				else if (args[i].equalsIgnoreCase("-config"))
-					fnameOptConfig = args[++i];
-				else if( args[i].equalsIgnoreCase("-debug") ) {					
-					ENABLE_DEBUG_MODE = true;
-				}
-				else if( args[i].equalsIgnoreCase("-gpu") ) {	
-					USE_ACCELERATOR = true;
-					if( args.length > (i+1) && !args[i+1].startsWith("-") ) {
-						String flag = args[++i];
-						if(flag.startsWith("force=")) {
-							String [] flagOptions = flag.split("=");
-							if(flagOptions.length == 2)
-								FORCE_ACCELERATOR = Boolean.parseBoolean(flagOptions[1]);
-							else
-								throw new DMLRuntimeException("Unsupported \"force\" option for -gpu:" + flag);
-						}
-						else {
-							throw new DMLRuntimeException("Unsupported flag for -gpu:" + flag);
-						}
-					}
-					GPUContext.getGPUContext(); // creates the singleton GPU context object. Return value ignored.
-				}
-				else if( args[i].equalsIgnoreCase("-python") ) {
-					parsePyDML = true;
-				}
-				else if (args[i].startsWith("-args") || args[i].startsWith("-nvargs")) {
-					namedScriptArgs = args[i].startsWith("-nvargs"); i++;
-					scriptArgs = new String[args.length - i];
-					System.arraycopy(args, i, scriptArgs, 0, scriptArgs.length); 
-					break;
-				}
-				else{
-					System.err.println("ERROR: Unknown argument: " + args[i]);
-					return false;
-				}
+			DMLOptions dmlOptions = parseCLArguments(args, options);
+
+			// String[] scriptArgs = null; //optional script arguments
+			// boolean namedScriptArgs = false;
+
+			STATISTICS 				= dmlOptions.stats;
+			STATISTICS_COUNT 	= dmlOptions.statsCount;
+			USE_ACCELERATOR 	= dmlOptions.gpu;
+			FORCE_ACCELERATOR = dmlOptions.forceGPU;
+			EXPLAIN 					= dmlOptions.explainType;
+			ENABLE_DEBUG_MODE = dmlOptions.debug;
+			SCRIPT_TYPE				= dmlOptions.scriptType;
```

Comment:
Can you fix the spacing in this section?  It would be better to just use spaces to align everything.